java-spiffe-core/src/main/java/io/spiffe/svid/x509svid/X509Svid.java

```diff
@@ -0,0 +1,169 @@
+     * @param privateKeyBytes private key as byte array
+     * @return a {@link X509Svid} parsed from the given certBytes and privateKeyBytes
+     * @throws X509SvidException if the given certsBytes or privateKeyBytes cannot be parsed
+     */
+    public static X509Svid parse(@NonNull final byte[] certsBytes, @NonNull final byte[] privateKeyBytes) throws X509SvidException {
+        return createX509Svid(certsBytes, privateKeyBytes);
+    }
+
+    /**
+     * @return the chain of certificates as an array of {@link X509Certificate}
+     */
+    public X509Certificate[] getChainArray() {
+        return chain.toArray(new X509Certificate[0]);
+    }
+
+    private static X509Svid createX509Svid(final byte[] certsBytes, final byte[] privateKeyBytes) throws X509SvidException {
+
+        List<X509Certificate> x509Certificates;
+        try {
+            x509Certificates = CertificateUtils.generateCertificates(certsBytes);
+        } catch (CertificateParsingException e) {
+            throw new X509SvidException("Certificate could not be parsed from cert bytes", e);
+        }
+
+        PrivateKey privateKey;
+        try {
+            privateKey = CertificateUtils.generatePrivateKey(privateKeyBytes);
+        } catch (InvalidKeySpecException | InvalidKeyException | NoSuchAlgorithmException e) {
+            throw new X509SvidException("Private Key could not be parsed from key bytes", e);
+        }
+
+        SpiffeId spiffeId;
+        try {
+            spiffeId = CertificateUtils.getSpiffeId(x509Certificates.get(0));
+        } catch (CertificateException e) {
+            throw new X509SvidException(e.getMessage(), e);
+        }
+
+        validatePrivateKey(privateKey, x509Certificates);
+        validateLeafCertificate(x509Certificates.get(0));
+
+        if (x509Certificates.size() > 1) {
+            validateSigningCertificates(x509Certificates.subList(1, x509Certificates.size()));
+        }
+
+        return new X509Svid(spiffeId, x509Certificates, privateKey);
+    }
+
+    private static void validateSigningCertificates(final List<X509Certificate> certificates) throws X509SvidException {
+        for (X509Certificate cert : certificates) {
+            if (!CertificateUtils.isCA(cert)) {
+                throw new X509SvidException("Signing certificate must have CA flag set to true");
+            }
+            if (!CertificateUtils.hasKeyUsageCertSign(cert)) {
+                throw new X509SvidException("Signing certificate must have 'keyCertSign' as key usage");
+            }
+        }
+    }
+
+    private static void validateLeafCertificate(final X509Certificate leaf) throws X509SvidException {
+        if (CertificateUtils.isCA(leaf)) {
+            throw new X509SvidException("Leaf certificate must not have CA flag set to true");
+        }
+        validateKeyUsage(leaf);
+    }
+
+    private static void validateKeyUsage(final X509Certificate leaf) throws X509SvidException {
```

Comment:
Rename to something like `validateKeyUsageOfLeafCertificate()` to be more explicit.

Reply:
Done.